### ISSUE TITLE
Does deliver upload debugging info to iTunes connect (dSYM)?

### ISSUE THREAD

**Issue by jeroenleenarts** [fastlane/deliver#227]
Quick question, docs and code don't give me a conclusive answer.

Do debug symbols get uploaded to ITC?

**Comment by KrauseFx** [comment]
@jeroenleenarts No, they don't. You'll need to take care of uploading them. I usually use 3rd party crash reporters like Crashlytics or Hockey.

**Comment by Panajev** [comment]
It should be possible and desirable to do so to get automatically symbolicated crash logs from App Store builds (according to the checkbox/message you get when uploading the build manually through Xcode). Is it in your roadmap? Can it be done by deliver?

[web [iPhone msg]]

**Comment by KrauseFx** [comment]
This feature is not on the roadmap. Feel free to submit a pull request if you have time working on it :+1:

**Comment by nixiphonecibuilder** [comment]
I haven't found an option to submit dSYM to ITC. Could you please help finding it?

**Comment by KrauseFx** [comment]
@nixiphonecibuilder This option is not implemented and is not available.

**Comment by artem-alexandrov** [comment]
You was talking about ITC right?

**Comment by appcoders** [comment]
The only thing thats needed is a additional folder in the IPA file called "Symbols" which contains extracted information from the dSYM files. Only problem - I did not find any parameter for xcodebuild to create such an archive.

**Comment by KrauseFx** [comment]
I need this feature now too. I already contacted Apple asking if there is a way to upload the `dsym` file using Transporter. Let's see if I get something back.

**Comment by appcoders** [comment]
Hi Felix,
Have a look at the bottom of
http://faq.sealedabstract.com/xcodeCI/

I managed to upload a IPA with a Symbols directory in the root containing
the symbols I created from
all contained app binaries.

Kind regards,
Markus

Felix Krause <notifications@github.com> schrieb am Sa., 11. Juli 2015 um
19:52:

**Comment by KrauseFx** [comment]
@appcoders looks interesting. I can see how to create a `.symbols` file, but I couldn't find how to upload it to iTunes Connect.

**Comment by Ahti** [comment]
@KrauseFx if I am not mistaken the other commenters are referring to adding a Symbols directory into the root of the IPA in which you upload the application.

I just tried exporting an IPA from Xcode and it contained a Symbols directory with two `.symbols` files in it. I have not yet tried uploading that to iTunes Connect to see wether it recognises the symbols, but this might not be a job for deliver but for the system that builds the IPA?

**Comment by appcoders** [comment]
Regarding the IPA bundle:
https://devforums.apple.com/message/1120211#1120211

You need to upload the IPA created this way.

Lukas is right - deliver should be able to upload this IPA as any IPA
before.


CSS IT Consulting
Dipl.Inf. (FH) Markus Schicker
Westendstr. 25
83301 Traunreut

Tel 08669 901024 ■ Fax 08669 1203069
Mobil 0151 229 46 922
Mail M.Schicker@cssitconsulting.de

2015-07-13 0:29 GMT+02:00 Lukas Stabe <notifications@github.com>:

**Comment by KrauseFx** [comment]
Okay, that's really interesting. So basically that should be part of the `ipa` action of `fastlane` where you can define whether you want to include symbols or not :+1:

**Comment by appcoders** [comment]
Yes indeed :-)

CSS IT Consulting
Dipl.Inf. (FH) Markus Schicker
Westendstr. 25
83301 Traunreut

Tel 08669 901024 ■ Fax 08669 1203069
Mobil 0151 229 46 922
Mail M.Schicker@cssitconsulting.de

2015-07-13 16:34 GMT+02:00 Felix Krause <notifications@github.com>:

**Comment by carlosefonseca** [comment]
This seems to work for placing a `dSYM` file inside the `ipa` on the said folder… I'll try using it soon to submit an app using `deliver`…

```ruby
require 'zip'

dsym_path = '/Users/carlos/Desktop/Test.app.dSYM'

ipa_file_name = "/Users/carlos/Desktop/Test.app.ipa"

Zip::File.open(ipa_file_name, Zip::File::CREATE) do |zipfile|
  zipfile.add("Symbols/Test.app.dSYM", dsym_path)
end
```

**Comment by CraigSiemens** [comment]
I looked at an `ipa` created by Xcode and the `Symbols` folder contains two `.symbols` files that are just binary data.

I decided to try a couple different ways of adding the `dSYM` to the `ipa` anyway.
**tl;dr** Nothing worked

---

##### Add `dSYM` to the `Symbols` folder in the `ipa`
Fails to upload to iTunes Connect with the error
> ERROR ITMS-90017: "This bundle is invalid.  The IPA format requires a top-level directory named Payload, containing only a .app bundle and optional plugins in a Plugins directory."

---

##### Add `dSYM.zip` to the `Symbols` folder in the `ipa`
Got an email for iTunes Connect that the app has issues
> Malformed Symbol Directory - The Symbol Directory contains invalid files [APPNAME.app.dSYM.zip]

**Comment by cameronehrlich** [comment]
:+1: this would be really handy

**Comment by CraigSiemens** [comment]
I created a custom fastlane action that does this. I posted it to https://github.com/fastlane/gym/issues/51 since it makes more sense for `gym` to support this since it builds the IPA.

**Comment by pedro380085** [comment]
+1

**Comment by adcooley** [comment]
+1 ya!

**Comment by RyogaK** [comment]
+1
